eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/Retry.java

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.exception.ErrorCondition;
+import com.azure.eventhubs.implementation.ClientConstants;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * An abstract representation of a policy to govern retrying of messaging operations.
+ */
+public abstract class Retry {
+
+    public static final Duration DEFAULT_RETRY_MIN_BACKOFF = Duration.ofSeconds(0);
+    public static final Duration DEFAULT_RETRY_MAX_BACKOFF = Duration.ofSeconds(30);
+    public static final int DEFAULT_MAX_RETRY_COUNT = 10;
+
+    private final AtomicInteger retryCount = new AtomicInteger();
+    private final int maxRetryCount;
+
+    /**
+     * Creates a new instance of Retry with the maximum retry count of {@code maxRetryCount}
+     *
+     * @param maxRetryCount The maximum number of retries allowed.
+     */
+    public Retry(int maxRetryCount) {
+        this.maxRetryCount = maxRetryCount;
+    }
+
+    /**
+     * Check if the existing exception is a retriable exception.
+     *
+     * @param exception An exception that was observed for the operation to be retried.
+     * @return true if the exception is a retriable exception, otherwise false.
+     */
+    public static boolean isRetriableException(Exception exception) {
+        return (exception instanceof AmqpException) && ((AmqpException) exception).isTransient();
+    }
+
+    /**
+     * Get default configured Retry.
+     *
+     * @return Retry which has all default property set up.
+     */
+    public static Retry getNoRetry() {
+        return new ExponentialRetry(
+            Duration.ZERO,
+            Duration.ZERO,
+            0);
+    }
+
+    /**
+     * Get default configured Retry.
+     *
+     * @return Retry which has all default property set up.
+     */
+    public static Retry getDefaultRetry() {
+        return new ExponentialRetry(
+            DEFAULT_RETRY_MIN_BACKOFF,
```

Comment:
nit: Any reason each parameter is on a new line? Does it reach the 120 characters/line limit?